### PR TITLE
Event, User, Attendance and Invitation Feature

### PR DESCRIPTION
**Pull request by Misori-simon**
### This pull request contains the following changes
- Created an Event Model with a description, a date, and a creator
 - - Added relevant controller methods for the related views
- Created a User Model with a name
 - - Added relevant controller methods for the related views
- Established a one to many relationships between user and events
 - - Added relevant controller methods for the related routes
- Created an Attendance Model as a many to many relations between users and events
 - Added relevant controller methods for the related views
- Created an Invitation Model as a many to many relations between users and events
- Created Tests for the User, Events, Attendance, and invitations
- Added styling for the project UI

### CODE REVIEW

File: app/controllers/events_controller.rb
Comment (teekaytech) [inline app/controllers/events_controller.rb]: You should use the  `#build` method on the association,
i.e. `@event = current_user.events.build(event_params)` ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/build_method_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/associations))

Note that `current_user` is a helper method, which you have to declare in your helper module.

File: app/controllers/events_controller.rb
Comment (teekaytech) [inline app/controllers/events_controller.rb]: You should list both past and upcoming events in the index action, i.e. 

`@past_events = Event.past_events` ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/past_upcoming_events_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/associations))
`@upcoming_events = Event.upcoming_events` ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/past_upcoming_events_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/associations))

You can then use these instance variables in your view.

File: app/controllers/users_controller.rb
Comment (teekaytech) [inline app/controllers/users_controller.rb]: lists events created by the user, i.e. 
`@created_events = current_user.events` ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/current_user_show_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/associations))

lists events previously attended by the user using event model scope, i.e. 
`@past_events = current_user.attended_events.past_events`

lists upcoming events previously attended/invited by the user using event model scope, i.e. 
`@upcoming_events = current_user.attended_events.upcoming_events`

**Review by teekaytech — CHANGES_REQUESTED** [review]
## Status: Changes Required :warning: :vertical_traffic_light: 

### Hello team :wave:

__Your project is awesome. You've done an amazing job so far :+1: 
However, some changes are required before it can be approved. Please check my comments below and fix them as appropriate.__

**[ REQUIRED CHANGES ]**

- [x] Update your readme file
- In the `To install and run` section, some information is missing:
  - After cloning, ask your user to navigate into the project directory
  - Also, some users need to run `yarn install` after running the `bundle install` command.
- [x] There should be absolutely NO logic in views. >Views are only for displaying information that became available to it from its corresponding controller not generating it.
      **Resources**
- [Where Do I Put My Code?](http://codefol.io/posts/Where-Do-I-Put-My-Code)
- [The Beginner's Guide to Rails Helpers](https://mixandgo.com/learn/the-beginners-guide-to-rails-helpers)
- [screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/section/no_logic_in_helpers_requirement.png) from Pathwright

- [x] Displaying the error messages for each action their corresponding views leads to unnecessary duplication of code. You can easily refactor by moving the error messages to the application layout so that it's available to all other views from one source.

- [x] You can add the `Attend` button to each event's show page. This allows any user to attend such an event by clicking on the button.

- [x] I left a couple of inline comments for you below. Kindly check and make appropriate changes.



* * *
**Please make the required changes and submit another review request when you are done.**

* * *

**Taofeek Olalere
[Twitter](https://twitter.com/ola_lere) | [Github](https://github.com/teekaytech) | @taofeekolalere on Slack**
__Happy Coding :smiley:__

**Comment by SuyashFowdar** [comment]
Hello @teekaytech ,

Thank you for reviewing our project.

I have a question concerning the Review No. 4, "_**You can add the Attend button to each event's show page. This allows any user to attend such an event by clicking on the button.**_".

In the Odin Project requirement, part **Event Attendance No. 9**, it tells us to allow only invited users to attend the events, so we added the button only when the user is invited. Should we add the **Attend** button for events even when the user can't attend it?

It would be of great help to us if this could be clarified. :+1:  Thank you very much. :smiley:

**Comment by teekaytech** [comment]
> Hello @teekaytech ,
> 
> Thank you for reviewing our project.
> 
> I have a question concerning the Review No. 4, "_**You can add the Attend button to each event's show page. This allows any user to attend such an event by clicking on the button.**_".
> 
> In the Odin Project requirement, part **Event Attendance No. 9**, it tells us to allow only invited users to attend the events, so we added the button only when the user is invited. Should we add the **Attend** button for events even when the user can't attend it?
> 
> It would be of great help to us if this could be clarified. +1 Thank you very much. smiley

Hello @krishnzzz,

Thanks for reaching out. As mentioned in your reference, it is said that adding this feature (i.e. allowing a user to invite other users) attracts extra credit.
Here's the screenshot from Odin:

This is the project statement:


You did well by implementing the invitation feature. I believe the project will be complete if you allow users to attend events by using the `Attend` button, based on the screenshot above. 

Happy coding!

**Review by ElijahTobs — APPROVED** [review]
# Status: Approved! :tada: 

## Congratulations! @Misori-simon and @krishnzzz  

Good job implementing the change as requested by the previous TSE.

## You can now merge and submit.

## Happy Coding :computer: 

_Feel free to reach out to me on slack_ **`@Elijah Ayandokun`**
  **Elijah O. Ayandokun**
[Github](https://github.com/ElijahTobs) | [LinkedIn](https://www.linkedin.com/in/ayandokunelijah/) | [Twitter](https://twitter.com/ElijahTobs)

**Pull request merged**
